poi-ooxml/src/main/java/org/apache/poi/xssf/streaming/SheetDataWriter.java

```diff
@@ -397,46 +397,51 @@ protected void outputEscapedString(String s) throws IOException {
             return;
         }
 
-        for (Iterator<String> iter = CodepointsUtil.iteratorFor(s); iter.hasNext(); ) {
-            String codepoint = iter.next();
+        for (int i = 0; i < s.length(); i++) {
+            final int codepoint = s.codePointAt(i);
+
             switch (codepoint) {
-                case "<":
+                case 60: // <
```

Comment:
could you test with `'<'`? - this should have similar performance but would be more readable

Reply:
Good idea. Done.